spec/stripe/input_data_validator_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+require 'stripe/input_data_validator'
+require 'spec_helper'
+
+module Stripe
+  RSpec.describe InputDataValidator do
+    RawInputData = Struct.new(:amount, :currency, :token, :email, :name)
+
+    let(:data) do
+      RawInputData.new(
+        '1000', 'usd', 'stripe_token', 'user@example.com', 'Name'
+      )
+    end
+    let(:result) { validate(data) }
+    let(:fields) { result.item }
+
+    describe 'Stripe required fields' do
+      it 'requires an amount' do
+        expect(fields[:amount]).to eq('1000')
```

Comment:
How does this tests check that it requires an amount?

It looks to me like it checks that it *stores* an amount. Likewise for all the other tests in this group.

Reply:
It's documenting the fields we're expecting, perhaps an alternative way would be to assert that we get an error if a required field is missing or invalid?

Reply:
what Ketan said.
also, the assertion of the error if a required field is missing is in the stripe gateway tests which test exception handling

Reply:
You test that the field is required later on in the spec, under validations. All this is doing it setting `amount` and then checking that it can read back `amount`.

Labelling this test "requires an amount" is entirely misleading in my opinion..